File: app/src/main/java/com/example/weatherapp/model/LocationWeatherDataRequest.java
package com.example.weatherapp.model;

import android.os.Parcel;
import android.os.Parcelable;

public class LocationWeatherDataRequest implements Parcelable {

    private final String    location;
    private final boolean   humidityParameter;
    private final boolean   pressureParameter;
    private final boolean   speedOfWindParameter;

    public LocationWeatherDataRequest(String location, boolean humidityParameter, boolean pressureParameter, boolean speedOfWindParameter) {
        this.location = location;
        this.humidityParameter = humidityParameter;
        this.pressureParameter = pressureParameter;
        this.speedOfWindParameter = speedOfWindParameter;
    }

    public String getLocation() {
        return location;
    }

    public boolean isHumidityParameter() {
        return humidityParameter;
    }

    public boolean isPressureParameter() {
        return pressureParameter;
    }

    public boolean isSpeedOfWindParameter() {
        return speedOfWindParameter;
    }

    public static final Creator<LocationWeatherDataRequest> CREATOR = new Creator<LocationWeatherDataRequest>() {

        @Override
        public LocationWeatherDataRequest createFromParcel(Parcel source) {

            String location             = source.readString();
            boolean humidityParameter   = source.readBoolean();
            boolean pressureParameter   = source.readBoolean();
            boolean speedOfWindParameter = source.readBoolean();

            return new LocationWeatherDataRequest(location,
                                                  humidityParameter,
                                                  pressureParameter,
                                                  speedOfWindParameter);
        }

        @Override
        public LocationWeatherDataRequest[] newArray(int size) {
            return new LocationWeatherDataRequest[size];
        }

    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(location);
        dest.writeBoolean(humidityParameter);
        dest.writeBoolean(pressureParameter);
        dest.writeBoolean(speedOfWindParameter);
    }

}

File: weatherapp/app/src/main/java/com/example/weatherapp/model/LocationWeatherDataRequest.kt
package com.example.weatherapp.model

import android.os.Parcel
import android.os.Parcelable

data class LocationWeatherDataRequest(
    val location: String,
    val humidityParameter: Boolean,
    val pressureParameter: Boolean,
    val speedOfWindParameter: Boolean) : Parcelable {

    private constructor(source: Parcel) : this(
        source.readString() as String,
        source.readValue(null) as Boolean,
        source.readValue(null) as Boolean,
        source.readValue(null) as Boolean
    )

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(location)
        dest.writeValue(humidityParameter);
        dest.writeValue(pressureParameter);
        dest.writeValue(speedOfWindParameter);
    }

    companion object CREATOR : Parcelable.Creator<LocationWeatherDataRequest> {
        override fun createFromParcel(source : Parcel): LocationWeatherDataRequest {
            return LocationWeatherDataRequest(source);
        }
        override fun newArray(size: Int): Array<LocationWeatherDataRequest?> {
            return arrayOfNulls(size)
        }
    }

}